probe-rs-tools/src/bin/probe-rs/cmd/cargo_embed/mod.rs

```diff
@@ -138,10 +140,14 @@ fn main_try(args: &[OsString], offset: UtcOffset) -> Result<()> {
 
     let _log_guard = setup_logging(None, config.general.log_level);
 
+    let mut registry = Registry::from_builtin_families();
+
     // Make sure we load the config given in the cli parameters.
     for cdp in &config.general.chip_descriptions {
         let file = File::open(Path::new(cdp))?;
-        probe_rs::config::add_target_from_yaml(file)
+        let family: ChipFamily = serde_yaml::from_reader(file)?;
+        registry
+            .add_target_family(family)
```

Comment:
Is there anything that speaks against `add_target_from_yaml` on the `Registry`?

Reply:
Not really, maybe a bit of separation of concerns. I can change this when I rebase.

Reply:
Okay :) I just thought we could have spared us the explicit dependency on serde_yaml :)